src/dp.rs

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Differential privacy (DP) primitives.
+use crate::field::FieldElement;
+use crate::vdaf::prg::SeedStream;
+use std::fmt::Debug;
+
+/// Positive rational number to represent DP parameters and noise distributions.
+#[allow(dead_code)]
+#[derive(Clone, Debug)]
+pub struct Rational {
+    numerator: u32,
+    denominator: u32,
+}
+
+impl Rational {
+    #[allow(dead_code)]
+    fn new(numerator: u32, denominator: u32) -> Self {
+        Rational {
+            numerator,
+            denominator,
+        }
+    }
+}
+
+/// Marker trait for differential privacy budgets (regardless of the specific accounting method).
+pub trait DifferentialPrivacyBudget {}
+
+/// Zero-concentrated differential privacy (zCDP) budget.
+#[allow(dead_code)]
+struct ZeroConcentratedDifferentialPrivacyBudget {
+    epsilon: Rational, // rho = (epsilon**2)/2
```

Comment:
nit: `rho` is defined twice in comments, here and on line 40 below. When updating this code we'll have to keep track of the definition of `rho` in two places. I think it's better to just maintain it in one place, in order to reduce the chance of comments becoming inconsistent.
```suggestion
    epsilon: Rational,
```